tools/monitoring/yaml-handler/parser.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// The yamlHandler is responsible for fetching, parsing config yaml file. It also allows user to
+// retrieve a particular record from the yaml.
+
+package yamlHandler
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+type AlertConditions struct {
+	JobNameRegex string `yaml:"job-name-regex"`
+	Occurrences  int
+	JobsAffected string `yaml:"jobs-affected"`
+	PrsAffected  string `yaml:"prs-affected"`
+	Period       int
+}
+
+type PatternSpec struct {
+	ErrorPattern string `yaml:"error-pattern"`
+	Hint         string
+	Alerts       []AlertConditions
+}
+
+type YamlFile struct {
```

Comment:
rename to ErrorPatternConfig?

Reply:
I want to emphasize that this struct is containing the entire yaml file, instead of part of the file. ErrorPatternConfig seems a bit vague and can apply to many structs